seed/serializers/rules.py

```diff
@@ -0,0 +1,105 @@
+# !/usr/bin/env python
+# encoding: utf-8
+"""
+:copyright (c) 2014 - 2020, The Regents of the University of California, through Lawrence Berkeley National Laboratory (subject to receipt of any required approvals from the U.S. Department of Energy) and contributors. All rights reserved.  # NOQA
+:author
+"""
+
+from rest_framework import serializers
+from seed.models.data_quality import Rule
+from seed.models import StatusLabel
+
+
+class RuleSerializer(serializers.ModelSerializer):
+    data_type = serializers.CharField(source='get_data_type_display', required=False)
+    status_label = serializers.PrimaryKeyRelatedField(
+        queryset=StatusLabel.objects.all(),
+        allow_null=True,
+        required=False
+    )
+    severity = serializers.CharField(source='get_severity_display', required=False)
+
+    class Meta:
+        model = Rule
+        fields = [
+            'condition',
+            'data_type',
+            'enabled',
+            'field',
+            'id',
+            'max',
+            'min',
+            'not_null',
+            'required',
+            'rule_type',
+            'severity',
+            'status_label',
+            'table_name',
+            'text_match',
+            'units',
+        ]
+
+    def validate_label(self, label):
+        """
+        Note: DQ Rules can be shared from parent to child but child orgs can
+        have their own labels. So, a Rule shouldn't be associated to Labels
+        from child orgs. In other words, Rule and associated Label should be
+        from the same org.
+        """
+        if label is not None and label.super_organization_id != self.instance.data_quality_check.organization_id:
+            raise serializers.ValidationError(
+                f'Label with ID {label.id} not found in organization, {self.instance.data_quality_check.organization.name}.'
+            )
+        else:
+            return label
+
+    def validate(self, data):
```

Comment:
This method was an attempt to replicate the manual validation checks happening in the previous DQ `save_data_quality_rules` endpoint.

Reply:
Hoping for feedback on the "two" validations performed here.

Could we make it easier to follow? Is it correct, or are there any logical holes or cases not covered? Is there a better way to implement this?

Reply:
@lainsworth8801 Might be worth your second opinion since you implemented the checks initially.